src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -419,6 +419,19 @@ public Page GetPage(long pageNumber)
             return GetPageInternal(pageNumber);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Page GetPage(long pageNumber, PagerRef pagerRef)
+        {
+            if (_disposed)
+                ThrowObjectDisposed();
+
+            // PERF: ReadOnly pages will not have a PagerRef, so we can still use the cache. 
+            if (_pageLocator.TryGetReadOnlyPage(pageNumber, out Page result))
```

Comment:
Since `EnsureMapped` is called by `AcquirePagePointerWithOverflowHandling` then I think we don't need to have this method (`public Page GetPage(long pageNumber, PagerRef pagerRef)`) because we don't have to call `EnsureMapped` in `VoronStream` - `PagerRef` becomes useless.